apps/docs-snippets/src/guide/transactions/transaction-response.test.ts

```diff
@@ -0,0 +1,104 @@
+    gasPrice = provider.getGasConfig().minGasPrice;
+  });
+
+  it('gets transaction response from contract call', async () => {
+    // #region transaction-response-1
+    // #import { TransactionResponse };
+
+    // Call a contract function
+    const call = await contract.functions.increment_count(15).call();
+
+    // Pick off the transaction response
+    const transactionResponse: TransactionResponse = call.transactionResponse;
+
+    // Retrieve the full transaction summary
+    const transactionSummary = await transactionResponse.getTransactionSummary();
+    // #endregion transaction-response-1
+
+    expect(call.transactionId).toEqual(transactionSummary.id);
+  });
+
+  it('gets transaction response from transaction request', async () => {
+    const scriptMainFunctionArguments = [1];
+    const resources = await wallet.getResourcesToSpend([{ amount: 1000, assetId: BaseAssetId }]);
+
+    // #region transaction-response-2
+    // #import { ScriptTransactionRequest, TransactionResponse };
+
+    // Instantiate the transaction request using a ScriptTransactionRequest and set
+    // the script main function arguments
+    const transactionRequest = new ScriptTransactionRequest({
+      script: scriptBytecode,
+      gasPrice,
+    });
+    transactionRequest.setData(scriptAbi, scriptMainFunctionArguments);
+
+    // Fund the transaction
+    transactionRequest.addResources(resources);
+
+    // Submit the transaction
+    const response: TransactionResponse = await wallet.sendTransaction(transactionRequest);
+
+    // Generate the transaction summary
+    const transactionSummary = await response.getTransactionSummary();
+    // #endregion transaction-response-2
+
+    expect(response.id).toEqual(transactionSummary.id);
+  });
+
+  it('gets transaction response from tx id', async () => {
+    const scriptMainFunctionArguments = [1];
+    const resources = await wallet.getResourcesToSpend([{ amount: 1000, assetId: BaseAssetId }]);
+
+    const transactionRequest = new ScriptTransactionRequest({
+      script: scriptBytecode,
+      gasPrice,
+    });
+    transactionRequest.setData(scriptAbi, scriptMainFunctionArguments);
+    transactionRequest.addResources(resources);
+    const response: TransactionResponse = await wallet.sendTransaction(transactionRequest);
+
+    // #region transaction-response-3
+    // #import { TransactionResponse };
+
+    // Get a transaction ID from the transaction request, this could have been stored from a
+    // previous transaction request. You will need the chain ID to obtain this.
+    const transactionId = transactionRequest.getTransactionId(provider.getChainId());
+
+    // Retrieve the transaction response from the transaction ID
+    const transactionResponse = await TransactionResponse.create(transactionId, provider);
+    
+    // Generate the transaction summary
+    const transactionSummary = await transactionResponse.getTransactionSummary();
+    // #endregion transaction-response-3
+
```

Comment:
This flow can be misleading because if the TX request was never submitted to the chain, getting its `TransactionResponse` will result in an error.

Here we want to use an ID from a TX that was already submitted to the node, like:

```ts
  ...

  const { id: transactionId } = await wallet.transfer(recipient, 100, assetId)

  // Retrieve the transaction response from the transaction ID
  const transactionResponse = await TransactionResponse.create(transactionId, provider);

  // Generate the transaction summary
  const transactionSummary = await transactionResponse.getTransactionSummary();
```

or even

```ts
  // id from an already processed TX
  const transactionId = '0x...'

  // Retrieve the transaction response from the transaction ID
  const transactionResponse = await TransactionResponse.create(transactionId, provider);
```

Reply:
So for this doc I give 3 examples. One for a contract call, one for a transaction request and this example was for an already processed TX. But I agree with your sentiment, so the latter example is more correct.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1961/commits/f8fd5dea744b62d2e384a7c4d0b0895ca574dbcf